awshelper/config_test.go

```diff
@@ -36,9 +38,61 @@ func TestAwsIsAddedInUserAgent(t *testing.T) {
 func TestAwsSessionValidationFail(t *testing.T) {
 	t.Parallel()
 
-	err := awshelper.ValidateAwsSession(&awshelper.AwsSessionConfig{
+	session, err := awshelper.CreateAwsSession(&awshelper.AwsSessionConfig{
 		Region:        "not-existing-region",
 		CredsFilename: "/tmp/not-existing-file",
 	}, options.NewTerragruntOptions())
+	require.NoError(t, err)
+
+	err = awshelper.ValidateAwsSession(session)
 	assert.Error(t, err)
 }
+
+// Test to validate cases when is not possible to read all S3 configurations
+// https://github.com/gruntwork-io/terragrunt/issues/2109
+func TestAwsNegativePublicAccessResponse(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name     string
+		response *s3.GetPublicAccessBlockOutput
+	}{
+		{
+			name: "nil-response",
+			response: &s3.GetPublicAccessBlockOutput{
+				PublicAccessBlockConfiguration: nil,
+			},
+		},
+		{
+			name: "legacy-bucket",
+			response: &s3.GetPublicAccessBlockOutput{
+				PublicAccessBlockConfiguration: &s3.PublicAccessBlockConfiguration{
+					BlockPublicAcls:       nil,
+					BlockPublicPolicy:     nil,
+					IgnorePublicAcls:      nil,
+					RestrictPublicBuckets: nil,
+				},
+			},
+		},
+		{
+			name: "false-response",
+			response: &s3.GetPublicAccessBlockOutput{
+				PublicAccessBlockConfiguration: &s3.PublicAccessBlockConfiguration{
+					BlockPublicAcls:       aws.Bool(false),
+					BlockPublicPolicy:     aws.Bool(false),
+					IgnorePublicAcls:      aws.Bool(false),
+					RestrictPublicBuckets: aws.Bool(false),
+				},
+			},
+		},
+	}
+	for _, testCase := range testCases {
```

Comment:
NIT: Try to keep this `tc`. It's what we use everywhere, and keeping it consistent is helpful.

Reply:
I don't mind to call it `tc`, but golint says, this is too short name.

Reply:
Which lint? I'm not seeing errors in the linter. We can add exceptions, given that it's our standard practice.

Reply:
Ah, it showed warning for `testCases`.
![strict_lint (67198) - gruntwork-ioterragrunt 2025-03-25 at 5 13 20 PM](https://github.com/user-attachments/assets/e52de382-bcb6-48f1-b524-5696001b435a)

Ok, I renamed `testCase` with `tc` throughout the code.

Reply:
Also renamed `tc := []strcut {` with `testCases := []strcut {`, and the `tt := range ...` with `tc := range ...`